### DISCUSSION TITLE
After Upgrade facing issue with 'Consent Popup' & 'custodian' org

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Lern/Community#158] [Issues] [answered]
Hello @reshmi-nair @AmiableAnil @vinukumar-vs,

**Sunbird Version - 4.10.0**

**Background -** 
We have upgraded the sunbird version from 4.4.1 to 4.10.0
Before upgrade we were not facing the following issue. 

**Issue Details -** 
1. After successfully logging in with a freshly registered user (generated from the backend, specifically Postman), the user is unable to accept the consent.

![explore](https://github.com/Sunbird-Lern/Community/assets/3881053/25ed5133-29a5-4ca5-88b8-b5b61e357acc)

After checking this box, the continue button is still disabled.becase in the code the condition is failed is 'updateConsent'

Note above issue not only for the 'custodianOrgId'.

**More Details**

Request URL

```
Request URL: https://host/learner/user/v1/consent/read
Request Method: POST
Status Code: 404 Not Found
```

Payload -

```
{
    "consent": {
        "filters": {
            "userId": "58bc46d2-f612-43bf-89e8-7e2e3e9fc60f",
            "consumerId": "0138778647572152323",
            "objectId": "0138778647572152323" 
        }
    }
}
```
Response

```
{
    "resmsgid": "2c2f338d165f103bce9a429654fdc734",
    "msgid": "2c2f338d165f103bce9a429654fdc734",
    "err": "USER_CONSENT_NOT_FOUND",
    "status": "USER_CONSENT_NOT_FOUND",
    "errmsg": "Requested user consent resource not found" 
} 
```
2. The issue we're having with custodian org is that when I register a user with custodian org and login with that user, I receive a discovery window with no options in the select field.

![discover content popup](https://github.com/Sunbird-Lern/Community/assets/3881053/f6aa40c3-3226-44cc-bbb7-d741389d2bdf)

3. The issue we're having with custodian org is that when we are enroll to the course with the custodian org user.

**Request URL**

```
Request URL: https://host/learner/user/v1/consent/read
Request Method: POST
Status Code: 404 Not Found
Remote Address: 20.244.62.154:443
Referrer Policy: strict-origin-when-cross-origin
```
Payload - 

```
{
    "consent": {
        "filters": {
            "userId": "106abf62-82fa-4c62-8300-baf3b36ccb4b",
            "consumerId": "0130152125009674241",
            "objectId": "do_113848746566967296117"
        }
    }
}
```

Response - 

```
response
{
    "id": "api.user.consent.read",
    "ver": "v1",
    "ts": "2023-09-11 05:31:32:046+0000",
    "params": {
        "resmsgid": "e44c976cf910e82c52e1d866d5dece9a",
        "msgid": "e44c976cf910e82c52e1d866d5dece9a",
        "err": "USER_CONSENT_NOT_FOUND",
        "status": "USER_CONSENT_NOT_FOUND",
        "errmsg": "Requested user consent resource not found"
    },
    "responseCode": "RESOURCE_NOT_FOUND",
    "result": {}
}
```

Update Consent Request URL -

```
Request URL: https://host/learner/user/v1/consent/update
Request Method: POST
Status Code: 400 Bad Request
Remote Address: 20.244.62.154:443
Referrer Policy: strict-origin-when-cross-origin
```

Response - 

```
{
    "id": "api.user.consent.update",
    "ver": "v1",
    "ts": "2023-09-11 05:31:36:367+0000",
    "params": {
        "resmsgid": "531b1f2b20429cc98d7195bf2d68a28d",
        "msgid": "531b1f2b20429cc98d7195bf2d68a28d",
        "err": "UOS_UCNSNTUPD0019",
        "status": "FAILED",
        "errmsg": "Please provide valid organisationid."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}
```

cc @Renuka-S, @vinod303, @coolbung, @Nithin-ns24 @sravani-64, @gandham-santhosh, @paritshivani

**Comment by mohitga** [comment]
@AmiableAnil  can you look into this issue?

**Comment by AmiableAnil** [comment]
@komalm 
What is the request body is passing in consent update API? In the response it is saying that `"errmsg": "Please provide valid organisationid."`

**Reply by komalm** [reply]
@AmiableAnil For Update API payload is the

```
{
    "request": {
        "consent": {
            "status": "ACTIVE",
            "userId": "106abf62-82fa-4c62-8300-baf3b36ccb4b",
            "consumerId": "0130152125009674241",
            "objectId": "do_113848746566967296117",
            "objectType": "Collection"
        }
    }
}
```

**Reply by komalm** [reply]
We have one question,

**Can we update/change the Custodian Org ID?**

Because the majority of the content and users in our case are created in the Custodian Org ID

cc @@sravani-64

**Comment by AmiableAnil** [comment]
@komalm Please refer this [code](https://github.com/Sunbird-Lern/userorg-service/blob/3d052efac894b4079376b213da55242174602364/service/src/main/java/org/sunbird/service/userconsent/impl/UserConsentServiceImpl.java#L117) 
If the consumerId is equals to custodianOrgId than this error is throwing.

**Comment by paritshivani** [comment]
@AmiableAnil ,
As all users are from current custodianOrg, both consumerId and custodianOrgId are same. but we cant change consumerId of users. Can we change custodianOrgId to any other orgnzation anyhow?


For more detailed discussion can we have a quick call today? this issue is critical and shows topper for us. we need to fix it in high priority.


cc @Renuka-S, @vinod303, @coolbung, @Nithin-ns2, @komalm , @sravani-64

**Comment by Krishnaj20** [comment]
@AmiableAnil @rajeevsathish - this is a blocker for Tekdi team. Please look into this today. If needed, pls get into a call.

@mohitga

**Reply by AmiableAnil** [reply]
@Krishnaj20 Scheduled a call at 2:30PM today.

**Reply by AmiableAnil** [reply]
@komalm @paritshivani Please update the call summary here.

**Reply by Krishnaj20** [reply]
thanks @AmiableAnil for taking out time to set up a call.

@renuka-s @vinod303, @coolbung, @Nithin-ns2, @komalm , @sravani-64 - as requested by @AmiableAnil  it is a good practice to summarize async conversations so that community and wider audience can benefit from this. Request that the call summary be posted here. Thanks

**Reply by Renuka-S** [reply]
@AmiableAnil @rajeevsatish Thank you so much for the call yesterday. As discussed over call we updated configurations and testing is in progress for impact. Will add the detailed updates and result in couple of days here. cc: @Krishnaj20

**Reply by Jayaprakash8887** [reply]
@Renuka-S Kindly update. Also, please mark the ticket as answered if there are no further concerns.

**Comment by paritshivani** [comment, marked as answer]
@AmiableAnil , @rajeevsatish, @Krishnaj20 , 
Tank you for your support.
Above issues got resolved after following steps mentioned bellow
1. For consent issue: we have this issue wile manual creation of user on postman. We need to compaire steps wit portal registration form
2. For discovery popup: We have redeployed the knowledge MW service and also added framework array to organisation
3. Custodian org issues, we changed thecustodian org id and custodian org channel wit system setting api
      API url: 'baseurl/api/data/v1/system/settings/set'
      Request body for custodian org Id: {
    "request": {
        "id": "custodianOrgId",
        "field": "custodianOrgId",
        "value": "{{organisationId}}"
    }
}
  Request body for custodian org cannel: {
    "request": {
        "id": "custodianOrgChannel",
        "field": "custodianOrgChannel",
        "value": "nulp-mainorg"
    }
}

**Reply by AmiableAnil** [reply]
@paritshivani Thanks for summarizing it. @komalm Please mark this thread as **Answered**.
